### ISSUE TITLE
Bug: The vector search is not supported for this elasticsearch version. Please, update the server to use this feature

### ISSUE THREAD

**Issue by getorca** [recognai/rubrix#2313]
**Describe the bug**
"The vector search is not supported for this elasticsearch version. Please, update the server to use this feature"

**To Reproduce**
1. run `wget -O docker-compose.yml https://raw.githubusercontent.com/argilla-io/argilla/main/docker-compose.yaml && docker-compose up -d`
2. run the script on the screen showing no dataset loaded:
```
# install datasets library with pip install datasets
import argilla as rg
from datasets import load_dataset

# load dataset from the hub
dataset = load_dataset("argilla/gutenberg_spacy-ner", split="train")

# read in dataset, assuming its a dataset for token classification
dataset_rg = rg.read_datasets(dataset, task="TokenClassification")

# log the dataset
rg.log(dataset_rg, "gutenberg_spacy-ner")
```
3. See the error:
```
Traceback (most recent call last):
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpcore/_exceptions.py", line 10, in map_exceptions
    yield
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpcore/backends/sync.py", line 28, in read
    return self._sock.recv(max_bytes)
ConnectionResetError: [Errno 104] Connection reset by peer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpx/_transports/default.py", line 60, in map_httpcore_exceptions
    yield
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpx/_transports/default.py", line 218, in handle_request
    resp = self._pool.handle_request(req)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpcore/_sync/connection_pool.py", line 253, in handle_request
    raise exc
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpcore/_sync/connection_pool.py", line 237, in handle_request
    response = connection.handle_request(request)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpcore/_sync/connection.py", line 90, in handle_request
    return self._connection.handle_request(request)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpcore/_sync/http11.py", line 112, in handle_request
    raise exc
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpcore/_sync/http11.py", line 91, in handle_request
    ) = self._receive_response_headers(**kwargs)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpcore/_sync/http11.py", line 155, in _receive_response_headers
    event = self._receive_event(timeout=timeout)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpcore/_sync/http11.py", line 191, in _receive_event
    data = self._network_stream.read(
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpcore/backends/sync.py", line 28, in read
    return self._sock.recv(max_bytes)
  File "/usr/lib/python3.8/contextlib.py", line 131, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpcore/_exceptions.py", line 14, in map_exceptions
    raise to_exc(exc)
httpcore.ReadError: [Errno 104] Connection reset by peer

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home[dir]/argilla/test_ds.py", line 12, in <module>
    rg.log(dataset_rg, "gutenberg_spacy-ner")
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/api.py", line 716, in wrapped_func
    return func(*args, **kwargs)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/api.py", line 755, in log
    return active_api().log(*args, **kwargs)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/api.py", line 695, in active_api
    __ACTIVE_API__ = Api()
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/api.py", line 159, in __init__
    self._user: User = users_api.whoami(client=self._client)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/sdk/users/api.py", line 23, in whoami
    response = client.get("/api/me")
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/sdk/client.py", line 128, in inner
    result = func(self, *args, **kwargs)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/sdk/client.py", line 139, in get
    response = self.__httpx__.get(
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpx/_client.py", line 1045, in get
    return self.request(
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpx/_client.py", line 821, in request
    return self.send(request, auth=auth, follow_redirects=follow_redirects)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpx/_client.py", line 908, in send
    response = self._send_handling_auth(
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpx/_client.py", line 936, in _send_handling_auth
    response = self._send_handling_redirects(
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpx/_client.py", line 973, in _send_handling_redirects
    response = self._send_single_request(request)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpx/_client.py", line 1009, in _send_single_request
    response = transport.handle_request(request)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpx/_transports/default.py", line 218, in handle_request
    resp = self._pool.handle_request(req)
  File "/usr/lib/python3.8/contextlib.py", line 131, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/httpx/_transports/default.py", line 77, in map_httpcore_exceptions
    raise mapped_exc(message) from exc
httpx.ReadError: [Errno 104] Connection reset by peer
(venv) lawrence@lawrence-office-2020:~/Projects/tutorials/argilla$ /home[dir]/argilla/venv/bin/python /home[dir]/argilla/test_ds.py
Using custom data configuration default-fb1202d521d42ac1
Found cached dataset parquet (/home/lawrence/.cache/huggingface/datasets/argilla___parquet/default-fb1202d521d42ac1/0.0.0/2a3b91fbd88a2c90d1dbbb32b460cf621d31bd5b05b934492fdef7d8d6f236ec)
  0%|                                                                                                                                 | 0/100 [00:00<?, ?it/s]Cannot log data ((), {'records': <argilla.client.datasets.DatasetForTokenClassification object at 0x7f9558e215b0>, 'name': 'gutenberg_spacy-ner', 'tags': None, 'metadata': None, 'chunk_size': 500, 'verbose': True})
Error of type <class 'argilla.client.sdk.commons.errors.GenericApiError'>
: Argilla server returned an error with http status: 500
Error details: [{'code': 'argilla.api.errors::GenericServerError', 'params': {'type': 'builtins.ValueError', 'message': 'The vector search is not supported for this elasticsearch version. Please, update the server to use this feature'}}]. (())
Traceback (most recent call last):
  File "/home[dir]/argilla/test_ds.py", line 12, in <module>
    rg.log(dataset_rg, "gutenberg_spacy-ner")
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/api.py", line 716, in wrapped_func
    return func(*args, **kwargs)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/api.py", line 755, in log
    return active_api().log(*args, **kwargs)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/api.py", line 310, in log
    return future.result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 444, in result
    return self.__get_result()
  File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
    raise self._exception
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/api.py", line 101, in __log_internal__
    raise ex
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/api.py", line 95, in __log_internal__
    return await api.log_async(*args, **kwargs)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/api.py", line 399, in log_async
    response = await async_bulk(
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/sdk/commons/api.py", line 100, in async_bulk
    return build_bulk_response(response, name=name, body=json_body)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/sdk/commons/api.py", line 115, in build_bulk_response
    return handle_response_error(response, name=name, body=body)
  File "/home[dir]/argilla/venv/lib/python3.8/site-packages/argilla/client/sdk/commons/errors_handler.py", line 61, in handle_response_error
    raise error_type(**error_args)
argilla.client.sdk.commons.errors.GenericApiError: Argilla server returned an error with http status: 500
Error details: [{'code': 'argilla.api.errors::GenericServerError', 'params': {'type': 'builtins.ValueError', 'message': 'The vector search is not supported for this elasticsearch version. Please, update the server to use this feature'}}]
```

**Expected behavior**
Datasets can be installed. 

**Additional context**
upgrading es to 8.6.1 resolves the issue, eg in docker-compose.yml:
```
  elasticsearch:
    image: docker.elastic.co/elasticsearch/elasticsearch:8.6.1
```
I'm going to guess this issue also exists in some other deployment methods.

**Comment by davidberenstein1957** [comment]
@getorca good catch! I believe `wget -O docker-compose.yml https://raw.githubusercontent.com/argilla-io/argilla/main/docker-compose.elasticsearch.yaml && docker-compose up -d` should fix your issue for now. 

@frascuchon do we want to remove the old `docker-compose`? or just bump the version?

**Comment by davidberenstein1957** [comment]
@rhoitjadhav this is about the ES versioning.

**Comment by dvsrepo** [comment]
I think we should provide a docker-compose with an ES version that supports vector search. @frascuchon wanted to do this gradually but I think it makes sense to upgrade the ES version to support vector search and maybe add a warning note about migration for those users that were using the previous docker-compose

**Comment by davidberenstein1957** [comment]
I agree. Also, in practice I am not sure if users continuously re-download the `docker-compose` or only do so when they actively want to update something.

**Comment by dvsrepo** [comment]
agree

**Comment by davidberenstein1957** [comment]
fixed here #2328
